presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixConfigurationInitializer.java

```diff
@@ -53,7 +53,7 @@
     private final int dataTransferServerPort;
 
     // Configs below are dependent on node joining the cluster
-    private boolean cacheNotReady = true;
+    private volatile boolean cacheNotReady = true;
```

Comment:
Let's reverse this variable, call it `cacheReady`

Reply:
How long lived are `Configuration` objects. Is there a potential for race condition where `Configuration` will be set without cache for minutes (even though cache initialize a second after)?

Reply:
> How long lived are `Configuration` objects. Is there a potential for race condition where `Configuration` will be set without cache for minutes (even though cache initialize a second after)?

I think there is a chance for that. See https://github.com/prestosql/presto/issues/3525.
So there is a possiblity that this whole `cacheReady` flag will have to go away together with mutability of the class.

Reply:
Ya, in worst case there is a possibility to have rubix enabled and initialized but not being used if all threadLocal configs are configured before rubix initialization. I think the `DynamicConfigurationProvider` approach mentioned there is the easiest one to solve this but it break the construct of only dynamic configurations injected using `DynamicConfigurationProvider` as rubix configs will be static once it is initialized.

Reply:
What is the downside of doing it that way? Technicaly speaking current Rubix config is dynamic. Just state change machine is simple :)

Reply:
Not 100% sure but not everything is on `DynamicConfigurationProvider` so I guess there is a reason to be cautious about what we put there. One reason could be to minimize the number Configuration object we create, with dynamic configurations there will be a new copy every time Configuration is asked for which will be at least once for every split in a node. Rubix config is somewhere in between DynamicConfigurationProvider and the static ConfigurationInitializers, it is dynamic until initialized and after that it is static.

Reply:
I think there are 2 approaches here:
1. We let queries run without caching until Rubix is initialized. Then we need to put initialization as part of `DynamicConfigurationProvider`. I don't think performance penalty is big looking at `getFileSystem` method callers.
2. We fail any request to `getFileSystem` until Rubix is initialized. Maybe we could make that part of `ConfigurationProvider`.

When I think about it more, 2) approach seems better as it's more explicit.